src/Chapter2.hs

```diff
@@ -842,7 +857,10 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [Int] -> [Int]
+rotate _ [] = []
+rotate 0 l = l
+rotate nb l = drop nb (take (length l + nb) (cycle l))
```

Comment:
Nice catch on the case when `nb` is zero 🔥 
There is a way to optimise it even more! The current solution does a little bit more than work when the given number is big and the list is small. You will cycle through the list a lot. you can check this case and reduce `nb` somehow 👍🏼